src/gf_nf4.h

```diff
@@ -205,127 +214,120 @@ inline T NF4<T>::neg(T a) const
 template <typename T>
 inline T NF4<T>::add(T a, T b) const
 {
-    uint32_t arr[this->n];
-
-    arr[0] = (narrow_cast<uint32_t>(a) + narrow_cast<uint32_t>(b)) % 65537;
+    scratch32[0] =
```

Comment:
As we discussed, `scratch16` and `scratch32` could lead to issues if this operation is executed in multi-threads.
Maybe a note about it is useful for now.

Reply:
The thing is, most of our class are not MT-safe (e.g: you can't call the method `fft` of `fft::Additive` in multiple thread => we will have a data race on the `mem` attribute, same for the `encode` of `fec::RsFnt`, and so on…)

I could add a comment for this one, but then for lot of other classes as well (I can do it, but in a dedicated PR).

The thing is, most of our class are not MT-safe at the instance level (you can do MT if each thread create its own object, but reusing an object accross multiple threads is not supported by QuadIron).

Reply:
okay, a dedicated PR might be a good for it